lib/after_commit_everywhere.rb

```diff
@@ -16,34 +16,58 @@ class NotInTransaction < RuntimeError; end
   delegate :after_commit, :before_commit, :after_rollback, to: AfterCommitEverywhere
   delegate :in_transaction?, to: AfterCommitEverywhere
 
+  # Causes {before_commit} and {after_commit} to raise an exception when
+  # called outside a transaction.
+  EXCEPTION = :exception
+  # Causes {before_commit} and {after_commit} to execute the given callback
+  # immediately when called outside a transaction.
+  EXECUTE = :execute
+  # Causes {before_commit} and {after_commit} to log a warning before calling
+  # the given callback immediately when called outside a transaction.
+  WARN_AND_EXECUTE = :warn_and_execute
+
   class << self
     # Runs +callback+ after successful commit of outermost transaction for
     # database +connection+.
     #
-    # If called outside transaction it will execute callback immediately.
+    # @param no_tx_action [Symbol] Determines the behavior of this function when
+    #   called without an open transaction.
+    #
+    #   Must be one of: {EXCEPTION}, {EXECUTE}, or {WARN_AND_EXECUTE}.
     #
     # @param connection [ActiveRecord::ConnectionAdapters::AbstractAdapter]
     # @param callback   [#call] Callback to be executed
     # @return           void
-    def after_commit(connection: ActiveRecord::Base.connection, &callback)
+    def after_commit(
+      no_tx_action = EXECUTE,
```

Comment:
Let's use keyword arguments for consistency and readability:
```suggestion
      no_tx_action: EXECUTE,
```
Also let's think about name of user-facing argument then. maybe something like `without_tx` or `outside_transaction`?

Reply:
`without_tx` sounds good to me.